tests/test_index.py

```diff
@@ -592,6 +592,144 @@ def test_hnsw_IP(self):
         assert np.allclose(Dref[mask, 0], Dhnsw[mask, 0])
 
 
+class TestNSG(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        unittest.TestCase.__init__(self, *args, **kwargs)
+        d = 32
+        nt = 0
+        nb = 1500
+        nq = 500
+        self.GK = 32
+
+        _, self.xb, self.xq = get_dataset_2(d, nt, nb, nq)
+
+    def make_knn_graph(self, metric):
+        n = self.xb.shape[0]
+        d = self.xb.shape[1]
+        index = faiss.IndexFlat(d, metric)
+        index.add(self.xb)
+        _, I = index.search(self.xb, self.GK + 1)
+        knn_graph = np.zeros((n, self.GK), dtype=np.int64)
+        for i in range(n):
```

Comment:
I don't understand what this loop is for, what about knn_graph = I[:, 1:] ?

Reply:
It considers the inner product distance. The distance between a vector and itself may not be the smallest. For example, given a zero vector `a` and a unit vector `b`, we have `d(a, b) = 0 < d(a, a) = 1`.